pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -0,0 +1,107 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Allows to connect to the cloud powered by Pasqal to run sequences."""
+from typing import Dict, List, Mapping, Optional, Union
+
+import sdk
+from numpy.typing import ArrayLike
+
+from pulser import Sequence
+from pulser.devices import Device
+from pulser.register import QubitId
+
+
+class PasqalCloud:
+    """Manager of the connection to the cloud powered by Pasqal.
+
+    The cloud connection enables to run sequences on simulators or on real
+    QPUs.
+
+    Args:
+        client_id: client_id of the API key you are holding for Pasqal
+            cloud.
+        client_secret: client_secret of the API key you are holding for
+            Pasqal cloud.
+        endpoints: Optionally, Pasqal cloud connection URLs.
+        webhook: Optionally, webhook.
+    """
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        endpoints: Optional[sdk.endpoints.Endpoints],
+        webhook: Optional[str] = None,
+    ):
+        """Initializes a connection to the cloud."""
+        self._sdk_connection = sdk.SDK(
+            client_id=client_id,
+            client_secret=client_secret,
+            endpoints=endpoints,
+            webhook=webhook,
+        )
+
+    def create_batch(
+        self,
+        seq: Sequence,
+        jobs: List[Dict[str, Union[ArrayLike, Mapping[QubitId, int]]]],
+        device_type: sdk.DeviceType = sdk.DeviceType.QPU,
+        configuration: Optional[sdk.Configuration] = None,
+        wait: bool = False,
+    ) -> sdk.Batch:
+        """Create a new batch and send it to the API.
+
+        For Iroise MVP, the batch must contain at least one job and will be
+        declared as complete immediately.
+
+        Args:
+            seq: Pulser sequence.
+            jobs: List of jobs to be added to the batch at creation.
+            device_type: The type of device to use, either an emulator or a QPU
+                If set to QPU, the device_type will be set to the one
+                stored in the serialized sequence.
+            configuration: A dictionary with extra configuration for the
+                emulators that accept it.
+            wait: Whether to wait for results to be sent back.
+
+        Returns:
+            Batch: The new batch that has been created in the database.
+        """
+        if device_type == sdk.DeviceType.QPU and not isinstance(
+            seq.device, Device
+        ):
+            raise TypeError(
+                "To be sent to a real QPU, the device of the sequence "
+                "must be a real device, instance of 'Device'."
+            )
+
+        self._sdk_connection.create_batch(
+            serialized_sequence=seq.serialize(),
```

Comment:
beware that some emulators do not / will not accept this serialization and need the abstract one. EMU_FREE and the QPUs accept the Python one.

We can probably merge it like this and leave that for when we can get the device configuration?

Reply:
@CdeTerra I think this should be addressed here. Otherwise it will fail when selecting EMU-TN. It is just a matter of adding an if statement:

```python
if device_type ==  sdk.DeviceType.EMU_TN:
    serialized_sequence = seq.abstract_representation()
else:
    serialized_sequence = seq.serialize()
```

Reply:
All emulators now support abstract representation. Only the QPU doesn't.
I guess it comes down to if @HGSilveri has made the abstract repr truly feature complete?

Reply:
Yes, it is feature complete now. Of course, it will be updated has Pulser gets new features.

Reply:
Well but cloud services still rely on both the abstract representation (for emulators) and the old sequence serializer (for QPUs) so I think this check is still needed for now.